File: PYTHON/01/arrays_medium_problems/kadane.py
class Solution:
    def function(self,arr):
        maxi = float('-inf')
        prev_sum = 0

        for i in range(len(arr)):
            prev_sum += arr[i]

            if prev_sum > maxi:
                maxi = prev_sum
            if prev_sum < 0:
                prev_sum = 0

        return maxi

if __name__ == '__main__':
    arr = [2, 3, 5, -2, 7, -4]
    obj = Solution()
    result = obj.function(arr)
    print(result)

File: JAVA/src/arrayMedium/kadane.java
package arrayMedium;

public class kadane {
    public static int function(int[] arr){
        int maxi = Integer.MIN_VALUE;
        int prev_sum = 0;

        for (int i =0; i < arr.length;i++){
            prev_sum += arr[i];
            if(prev_sum > maxi){
                maxi = prev_sum;
            }
            if (prev_sum < 0){
                prev_sum = 0;
            }
        }
        return maxi;
    }
    public static void main(String[] args) {
        int[] arr = {-3,-2,-2,-3};
        int[] arr2 = {2, 3, 5, -2, 7, -4};
        int result = function(arr);
        System.out.println(result);
        int res2 = function(arr2);
        System.out.println(res2);
    }
}
